src/_pytest/cacheprovider.py

```diff
@@ -269,7 +277,12 @@ def pytest_collectreport(self, report):
         else:
             self.lastfailed[report.nodeid] = True
 
-    def pytest_collection_modifyitems(self, session, config, items):
+    @pytest.hookimpl(hookwrapper=True, tryfirst=True)
+    def pytest_collection_modifyitems(
+        self, config: Config, items: List[nodes.Item]
+    ) -> Generator[None, None, None]:
+        yield
```

Comment:
Why is this change (and in `pytest_collection_modifyitems`) needed?

Reply:
i believe this is to ensure the actual code runs as late as possible

in this case this means, its running after most normal hook-wrappers finish (only the last bit of a earlier hook-wrapper could interact with it after that